src/types.ts

```diff
@@ -80,12 +226,44 @@ type OnlyValidatedFlatOptions = {
   pkg: string;
 };
 
+/**
+ * Utility type that represents an `UnValidated` type after validation,
+ * replacing any properties in the unvalidated type that also exist in the
+ * `Validated` type with the validated versions.
+ *
+ * @template UnValidated - The type representing the unvalidated form.
+ * @template Validated - The type representing the validated form.
+ */
 type ValidatedForm<UnValidated, Validated> = Omit<UnValidated, keyof Validated> & Validated;
 
-export type ValidatedBaseSignOptions = Readonly<ValidatedForm<BaseSignOptions, OnlyValidatedBaseSignOptions>>;
-export type SignOptions = Readonly<OnlySignOptions & BaseSignOptions>;
-export type ValidatedSignOptions = Readonly<ValidatedForm<OnlySignOptions, OnlyValidatedSignOptions> &
-  ValidatedBaseSignOptions>;
-export type FlatOptions = Readonly<OnlyFlatOptions & BaseSignOptions>;
-export type ValidatedFlatOptions = Readonly<ValidatedForm<OnlyFlatOptions, OnlyValidatedFlatOptions> &
-  ValidatedBaseSignOptions>;
+type ValidatedBaseSignOptions = Readonly<
+  ValidatedForm<BaseSignOptions, OnlyValidatedBaseSignOptions>
+>;
+
+type _SignOptions = Readonly<OnlySignOptions & BaseSignOptions>;
+
+/**
+ * Options for codesigning a packaged `.app` bundle.
+ */
+export interface SignOptions extends _SignOptions {}
```

Comment:
If we prefer keeping this exported as a `type`, we can instead tag `_SignOptions` as an `@interface`.

The tradeoff here is that TypeDoc doesn't detect `@interface` properly for function signatures, so the doc output for `signAsync()` will not expand the parameters properly.

<img width="450" alt="image" src="https://github.com/electron/osx-sign/assets/16010076/5fff21ab-edf3-4ede-ac8d-c8af3ac48582">

Reply:
I'm in favor of exporting it as an interface if it means signAsync() will have correct params